chore: MKT-17450 Proofread popular Kotlin docs 5 (#2488)
JetBrains/kotlin-web-site: docs/topics/gradle.md

> [//]: # (title: Gradle)

Reviewer: I've checked.

> [//]: # (title: Gradle)

Author: Thanks, I've committed your suggestions.

In order to build a Kotlin project with Gradle, you should [apply the Kotlin Gradle plugin to your project](#plugin-and-versions)
and [configure the dependencies](#configuring-dependencies).

## Plugin and versions

> Apply the Kotlin Gradle plugin by using [the Gradle plugins DSL](https://docs.gradle.org/current/userguide/plugins.html#sec:plugins_block).

Reviewer: ```suggestion
Apply the Kotlin Gradle plugin by using the [Gradle plugins DSL](https://docs.gradle.org/current/userguide/plugins.html#sec:plugins_block).
```

### Kotlin and Java sources for JavaScript

This plugin only works for Kotlin files, so it is recommended that you keep Kotlin and Java files separate (if the

> project contains Java files). If you don't store them separately , specify the source folder in the `sourceSets` block:

Reviewer: ```suggestion
project contains Java files). If you don't store them separately, specify the source folder in the `sourceSets` block:
```

### Dependency on the standard library

A dependency on the standard library (`stdlib`) is added automatically to each source set. The version
of the standard library used is the same as the version of the Kotlin Gradle plugin.

For platform-specific source sets, the corresponding platform-specific variant of the library is used, while a common standard
library is added to the rest. The Kotlin Gradle plugin will select the appropriate JVM standard library depending on
the `kotlinOptions.jvmTarget` [compiler option](#compiler-options) of your Gradle build script.

If you declare a standard library dependency explicitly (for example, if you need a different version), the Kotlin Gradle

> plugin won’t override it or add a second standard library.

Reviewer: ```suggestion
plugin won't override it or add a second standard library.
```

[Learn how to test code using JUnit on the JVM](jvm-test-using-junit.md).

If you need to use a different JVM test framework, disable automatic testing framework selection by

> adding the line `kotlin.test.infer.jvm.variant=false` to the project’s `gradle.properties` file.

Reviewer: ```suggestion
adding the line `kotlin.test.infer.jvm.variant=false` to the project's `gradle.properties` file.
```

After doing this, add the framework as a Gradle dependency.

If you had used a variant of `kotlin("test")` in your build script explicitly and project build stopped working with
a compatibility conflict,
see [this issue in the Compatibility Guide](compatibility-guide-15.md#do-not-mix-several-jvm-variants-of-kotlin-test-in-a-single-project).

### Set a dependency on a kotlinx library

If you use a kotlinx library and need a platform-specific dependency, you can use platform-specific variants

> of libraries with suffixes such as `-jvm` or `-js`, for example, `kotlinx-coroutines-core-jvm`. You can also use the library’s

Reviewer: ```suggestion
of libraries with suffixes such as `-jvm` or `-js`, for example, `kotlinx-coroutines-core-jvm`. You can also use the library's
```

</tabs>

If you use a multiplatform library and need to depend on the shared code, set the dependency only once, in the shared

> source set. Use the library’s base artifact name, such as `kotlinx-coroutines-core` or `ktor-client-core`.

Reviewer: ```suggestion
source set. Use the library's base artifact name, such as `kotlinx-coroutines-core` or `ktor-client-core`.
```

}
```

</tabs>

### Set dependencies at the top level

Alternatively, you can specify the dependencies at the top level, using the following pattern for the configuration names:
`<sourceSetName><DependencyType>`. This can be helpful for some Gradle built-in dependencies, like `gradleApi()`, `localGroovy()`,

> or `gradleTestKit()`, which are not available in the source sets’ dependency DSL.

Reviewer: ```suggestion
or `gradleTestKit()`, which are not available in the source sets' dependency DSL.
```

## Incremental compilation

The Kotlin Gradle plugin supports incremental compilation. Incremental compilation tracks changes to source files between
builds so only files affected by these changes are compiled.

Incremental compilation is supported for Kotlin/JVM and Kotlin/JS projects and is enabled by default.

There are several ways to switch off incremental compilation:

> * `kotlin.incremental=false` for Kotlin/JVM

Reviewer: ```suggestion
* `kotlin.incremental=false` for Kotlin/JVM.
```

> * `kotlin.incremental.js=false` for Kotlin/JS projects

Reviewer: ```suggestion
* `kotlin.incremental.js=false` for Kotlin/JS projects.
```

>
{type="note"}

The Kotlin plugin uses the [Gradle configuration cache](https://docs.gradle.org/current/userguide/configuration_cache.html),
which speeds up the build process by reusing the results of the configuration phase.

> Consult the [Gradle documentation](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:usage)

Reviewer 2: Not sure about the change from 'See' to 'Consult'. It's more formal; ans 'For more information, see' is a standart wording for such cases

</tabs>

Here is a complete list of options for Gradle tasks:

### Attributes common to JVM, JS, and JS DCE

| Name | Description | Possible values |Default value |
|------|-------------|-----------------|--------------|
| `allWarningsAsErrors` | Report an error if there are any warnings |  | false |

> | `suppressWarnings` | Don’t generate warnings |  | false |

Reviewer: ```suggestion
| `suppressWarnings` | Don't generate warnings |  | false |
```
